src/calibre/utils/icu_calibre_utils.h

```diff
@@ -22,25 +22,26 @@
 #include <unicode/unorm2.h>
 #include <unicode/ubrk.h>
 
-#if PY_VERSION_HEX >= 0x03030000
-#error Not implemented for python >= 3.3
+#if PY_VERSION_HEX < 0x03030000 && PY_VERSION_HEX > 0x03000000
+#error Not implemented for python 3.0 to 3.2
 #endif
 
+#if PY_VERSION_HEX < 0x03000000
 #define MIN(x, y) ((x)<(y)) ? (x) : (y)
 #define IS_HIGH_SURROGATE(x) (0xd800 <= x && x <= 0xdbff)
 #define IS_LOW_SURROGATE(x) (0xdc00 <= x && x <= 0xdfff)
 
 // Roundtripping will need to be implemented differently for python 3.3+ where strings are stored with variable widths
 
 #ifndef NO_PYTHON_TO_ICU
-static UChar* python_to_icu(PyObject *obj, int32_t *osz, uint8_t do_check) {
+static UChar* python_to_icu(PyObject *obj, int32_t *osz) {
     UChar *ans = NULL;
     Py_ssize_t sz = 0;
 #ifdef Py_UNICODE_WIDE
     UErrorCode status = U_ZERO_ERROR;
 #endif
 
-    if (do_check && !PyUnicode_CheckExact(obj)) {
+    if (!PyUnicode_CheckExact(obj)) {
```

Comment:
Wont this change prevent using unicode sub-classes?

Reply:
`do_check` was never anything but 1, so I removed it. I wasn't exactly sure why `PyUnicode_CheckExact` was used instead of `PyUnicode_Check`, but I figured that there was some reason, so I left it alone. I can change it to `PyUnicode_Check`

Reply:
ah well, probably best to leave it alone in that case, unless we have some actual reason to worry about unicode sub-classes. I confess I dont recall why I used checkexact there either :)